UI/script.js

```diff
@@ -1,79 +1,450 @@
-postArray = [
-    {
-        id: '1',
-        descriprion: 'Description...........',
-        createdAt: new Date('2019-03-01T23:00:00'),
-        author: 'ibelous',
-        path: '/images/img2.jpg',
-        display: true,
-        hashtags:['some', 'hash', 'tags'],
-        likers:['User1','User2','User3','User4']
-    },
-    {
-        id: '2',
-        descriprion: '12 312 3wa asf 23 sdf',
-        createdAt: new Date('2019-03-01T23:00:00'),
-        author: 'UserName',
-        path: '/images/img1.jpg',
-        display: true,
-        hashtags:['aaaa'],
-        likers:['User1','User2','User3','User4']
-    }
-];
-console.log('started');
-var modul = (function () {
-    function addPhotoPost(post){
-        if(validatePhotoPost(post)){
-            postArray.push(post);
-            console.log('added post');
-            return true;
-            }
-        console.log('invalid post');
-        return false;
-        }
-    function validatePhotoPost(post) {
-        if (!post.hasOwnProperty("id") || typeof post.id != "string" ||
-            !post.hasOwnProperty("description") || post.description.length >= 200 || typeof post.description != "string" ||
-            !post.hasOwnProperty("createdAt") ||
-            !post.hasOwnProperty("author") || typeof post.author != "string" ||
-            !post.hasOwnProperty("path") || typeof post.path != "string" ||
-            !post.hasOwnProperty("likes")) {
-            console.log("Post with id " + post.id + " is not valid.");
-            return false;
-        }
-        else {
-            console.log("Post with id " + post.id + " is valid.");
-            return true;
+class PostModel {
+  constructor(posts) {
+    this._photoPosts = posts;
+  }
+
+  _isIntersect(postTags, configTags) {
+    let num = 0;
+    postTags.forEach((item) => {
+      configTags.forEach((tag) => {
+        if (item === tag) {
+          num += 1;
         }
+      });
+    });
+    return num === configTags.length;
+  }
+
+  addAll(posts) {
+    const notValid = [];
+    posts.forEach((post) => {
+      if (this.addPhotoPost(post) === false) {
+        notValid.push(post);
+      }
+    });
+    return notValid;
+  }
+
+  getPhotoPosts(skip = 0, count = 10, filterConfig = PostModel._DEFAULT_FILTER) {
+    const filtratedPosts = this._photoPosts.filter(
+      post => post.createdAt.getTime() >= filterConfig.dateFrom.getTime()
```

Comment:
It's better to separate or format such things.
That predicate is very difficult to read.